### PR TITLE
Use commands config in help system

### LINKED ISSUE #39: Consider Config File for Commands
We currently have several places to change (`README`, `bin/_help`, `bin/[command]`) when we add or remove a command. Perhaps we should have a parsable configuration file containing the commands, their description etc... We can also use the configuration to determine of the command is a ["standard" command](https://github.com/technekes/nib/pull/37) and just run it.

**config/commands.json**

```json
{
  "attach": {
    "type": "custom",
    "description": "Attach an interactive shell session to a running container"
  },
  "bundle": {
    "type": "wrap",
    "description": "Run bundle for the given service"
  },
  "bootstrap": {
    "type": "custom",
    "description": "Runs the bootstrap script for the requested app (or all apps if 'apps' is specified)"
  },
  "console": {
    "type": "custom",
    "description": "Start a REPL session for the given service"
  },
  "debug": {
    "type": "custom",
    "description": "Connect to a running byebug server for a given service"
  },
  "rails": {
    "type": "wrap",
    "description": "Run the rails command for the given service"
  },
  "rake": {
    "type": "wrap",
    "description": "Run the rake command for the given service"
  },
  "rspec": {
    "type": "wrap",
    "description": "Runs the rspec command for the given service"
  },
  "rubocop": {
    "type": "wrap",
    "description": "Runs the rubocop command for the given service"
  },
  "run": {
    "type": "custom",
    "description": "Wraps normal 'docker-compose run' to ensure that --rm is always passed"
  },
  "shell": {
    "type": "custom",
    "description": "Start a shell session in a one-off service container"
  },
  "update": {
    "type": "custom",
    "description": "Download the latest version of the nib tool"
  }
}
```

Full credit for this idea goes to @blankenshipz, I'm just the scribe.

### PR DESCRIPTION
**Pull request by johnallen3d**
Instead of using a hard coded array of commands (which needs to be maintained separately) we now use the commands config.

This would resolve #39.

### CODE REVIEW

**Comment by johnallen3d** [comment]
👀  @blankenshipz - it ended up being a little simpler than we thought. 😄

File: bin/_help [hunk 1]
```diff
@@ -22,7 +22,8 @@ print_services() {
 }
 
 print_command_instructions() {
-  echo "  " $(printf -- '%s\n' "${COMMANDS[@]}" | grep "${COMMAND}|" | awk -F '|' '{print $3}')
+  echo $(commands_json) | \
+    jq -r '.[] | select(.name == "attach") | .description'
```
Comment (blankenshipz) [inline bin/_help]: Should there be a $1 here somewhere?

**Comment by blankenshipz** [comment]
:+1: :shipit:

File: bin/_help (hunk 1 above)
Comment (johnallen3d) [inline bin/_help]: Latest commit should resolve this issue.

**Comment by johnallen3d** [comment]
Merged offline.

**Pull request closed without merging**